src/test/ui/consts/offset_of.rs

```diff
@@ -0,0 +1,22 @@
+// check-pass
+
+#![feature(const_raw_ptr_deref)]
+#![feature(const_raw_ptr_to_usize_cast)]
+#![feature(const_ptr_offset_from)]
+#![feature(ptr_offset_from)]
+
+struct Struct {
+    field: (),
+}
+
+pub const OFFSET: usize = {
+    let uninit = std::mem::MaybeUninit::<Struct>::uninit();
+    let base_ptr: *const Struct = &uninit as *const _ as *const Struct;
+    let field_ptr = unsafe { &(*base_ptr).field as *const () as *const u8 };
+    let offset = unsafe { field_ptr.offset_from(base_ptr as *const u8) };
```

Comment:
Please add more tests -- in particular, hitting every different failure mode (different allocation, overflow, not a multiple of the size, ...).

Reply:
And please also add a working test for a non-0 offset, just to make it a bit more interesting.

Reply:
Added all tests I could think of